Sources/Sentry/SentryNSDataTracker.m

```diff
@@ -0,0 +1,186 @@
+{
+    @synchronized(self) {
+        self.isEnabled = NO;
+    }
+}
+
+- (BOOL)measureNSData:(NSData *)data
+          writeToFile:(NSString *)path
+           atomically:(BOOL)useAuxiliaryFile
+               method:(BOOL (^)(NSString *, BOOL))method
+{
+    id<SentrySpan> span = [self startTrackingWritingNSData:data filePath:path];
+
+    BOOL result = method(path, useAuxiliaryFile);
+
+    if (span != nil) {
+        [self finishTrackingNSData:data span:span];
+    }
+    return result;
+}
+
+- (BOOL)measureNSData:(NSData *)data
+          writeToFile:(NSString *)path
+              options:(NSDataWritingOptions)writeOptionsMask
+                error:(NSError **)error
+               method:(BOOL (^)(NSString *, NSDataWritingOptions, NSError **))method
+{
+    id<SentrySpan> span = [self startTrackingWritingNSData:data filePath:path];
+
+    BOOL result = method(path, writeOptionsMask, error);
+
+    if (span != nil) {
+        [self finishTrackingNSData:data span:span];
+    }
+
+    return result;
+}
+
+- (NSData *)measureNSDataFromFile:(NSString *)path method:(NSData * (^)(NSString *))method
+{
+    id<SentrySpan> span = [self startTrackingReadingFilePath:path];
+
+    NSData *result = method(path);
+
+    if (span != nil) {
+        [self finishTrackingNSData:result span:span];
+    }
+
+    return result;
+}
+
+- (NSData *)measureNSDataFromFile:(NSString *)path
+                          options:(NSDataReadingOptions)readOptionsMask
+                            error:(NSError **)error
+                           method:(NSData * (^)(NSString *, NSDataReadingOptions, NSError **))method
+{
+    id<SentrySpan> span = [self startTrackingReadingFilePath:path];
+
+    NSData *result = method(path, readOptionsMask, error);
+
+    if (span != nil) {
+        [self finishTrackingNSData:result span:span];
+    }
+
+    return result;
+}
+
+- (NSData *)measureNSDataFromURL:(NSURL *)url
+                         options:(NSDataReadingOptions)readOptionsMask
+                           error:(NSError **)error
+                          method:(NSData * (^)(NSURL *, NSDataReadingOptions, NSError **))method
+{
+
+    // We dont track reads from a url that is not a file url
+    // because these reads are handled by NSURLSession and
+    // SentryNetworkTracker will create spans in these cases.
+    if (![url.scheme isEqualToString:NSURLFileScheme])
+        return method(url, readOptionsMask, error);
+
+    id<SentrySpan> span = [self startTrackingReadingFilePath:url.path];
+
+    NSData *result = method(url, readOptionsMask, error);
+
+    if (span != nil) {
+        [self finishTrackingNSData:result span:span];
+    }
+
+    return result;
+}
+
+- (nullable id<SentrySpan>)spanForPath:(NSString *)path
+                             operation:(NSString *)operation
+                                  size:(NSUInteger)size
+{
+    @synchronized(self) {
+        if (!self.isEnabled || ![self shouldTrackPath:path])
+            return nil;
+    }
```

Comment:
`m`: It's good practice to keep the scope of the lock as minimum as possible. `shouldTrackPath:path` doesn't need to be inside the lock.
```suggestion
    @synchronized(self) {
        if (!self.isEnabled)
            return nil;
    }
    
    if (![self shouldTrackPath:path]) {
        return nil;
    }
```